### PR TITLE
feat(swc/cli): implements few compile flags

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

Update few more flags for the compile command.

**Related issue (if exists):**

https://github.com/swc-project/swc/issues/4017

### CODE REVIEW

**Comment by kwonoj** [comment]
Side note, I haven't thought deeply but implementing `config` seems non trivial work since we can't dynamically access rust struct's property with string key indexer as if we did in js object. I spent some time for it, and trying couple of approaches to make it work.

File: crates/swc_cli/src/commands/compile.rs [hunk 1]
```diff
@@ -271,6 +253,36 @@ struct InputContext {
...
 #[swc_trace]
 impl CompileOptions {
+    fn build_transform_options(&self, file_path: &Option<&Path>) -> anyhow::Result<Options> {
+        let base_options = Options::default();
+        let base_config = Config::default();
+
+        let config_file = if let Some(config_file_path) = &self.config_file {
+            let config_file_contents = fs::read(config_file_path)?;
+            serde_json::from_slice(&config_file_contents)
+                .map_err(|e| anyhow::anyhow!("{}", e))
```
Comment (kdy1) [inline crates/swc_cli/src/commands/compile.rs:263]: Is this required?
I think `.context("")` is enough to convert an error to `anyhow::Error`

Comment (kwonoj) [reply crates/swc_cli/src/commands/compile.rs]: yes, looks like old oversight.

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

---

swc-bump:
 - swc_cli

**Review by kdy1 — APPROVED** [review]

**Pull request merged**